### PR TITLE
feat: Kds 607 avoid fatal error when rte data doesnt match schema

### PR DESCRIPTION
**Pull request by JakePitman**
## Why
Current error state is jarring and confusing to the user.
Currently no ability for consumers to log errors.

## What
- Adds an optional `onDataError` prop that gets fired off when RichTextEditor receives bad data
- Adds an optional `dataError` prop (React Element) that gets displayed instead of the editor, when it receives bad data

### CODE REVIEW

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx [hunk 1]
```diff
@@ -56,25 +58,41 @@ export const RichTextEditor: React.VFC<RichTextEditorProps> = props => {
     classNameOverride,
     controls,
     rows = 3,
+    dataError = <p>Something went wrong</p>,
```
Comment (JakePitman) [inline packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx:61]: I'm wondering if it makes sense to have a default, and if so - what would be best?

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx [hunk 2]
```diff
@@ -30,6 +30,8 @@ export interface BaseRichTextEditorProps
    * The editable area will autogrow, so this only affects the component when the content doesn't exceed this height.
    */
   rows?: EditorRows
+  dataError?: React.ReactElement
```
Comment (JakePitman) [inline packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx:33]: Would ReactElement be best here? Or ReactNode? Do we want consumers to pass in raw strings, or should we restrict it to elements only?

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx [hunk 3]
```diff
@@ -56,25 +58,41 @@ export const RichTextEditor: React.VFC<RichTextEditorProps> = props => {
...
+              })
+            : null,
+          schema,
+          plugins: getPlugins(controls, schema),
+        }),
+        { "aria-labelledby": labelId, role: "textbox" }
+      )
+    } catch {
+      return new Error("Bad data error")
+    }
```
Comment (mcwinter07) [inline packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx:86]: Looks good 👍

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx [hunk 4]
```diff
@@ -56,25 +58,41 @@ export const RichTextEditor: React.VFC<RichTextEditorProps> = props => {
...
+  })()
+
+  if (useRichTextEditorResult instanceof Error) {
+    onDataError && onDataError()
+    return dataError
```
Comment (mcwinter07) [inline packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx:91]: We could render the inline notification here and pass in the dataError in as a child

**Comment by mcwinter07** [comment]
Just hand a thought, we might also want to create a story in the RichTextEditor.stories.tsx

File: packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx [hunk 5]
```diff
@@ -56,25 +59,45 @@ export const RichTextEditor: React.VFC<RichTextEditorProps> = props => {
...
+      )
+    } catch {
+      return new Error("Bad data error")
+    }
+  })()
+
+  if (useRichTextEditorResult instanceof Error) {
+    onDataError && onDataError()
+    return (
+      <InlineNotification title="Error" type="negative">
```
Comment (mcwinter07) [inline packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx:93]: Wondering if we should throw a `persistent` prop on the notification so it's not dismissible?

File: packages/rich-text-editor/docs/dummyMalformedContent.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "type": "plaragraph",
```
Comment (mcwinter07) [inline packages/rich-text-editor/docs/dummyMalformedContent.json:3]: so happy you went with plaragraph in the end 😄

**Review by mcwinter07 — APPROVED** [review]
Awesome stuff, the changes look hunky dory!  🦖 merge in good health

**Pull request merged**